test/gtest/uct/ib/test_devx.cc

```diff
@@ -48,4 +48,202 @@ UCS_TEST_P(test_devx, dbrec)
+                          (memh->exported_lkey & 0xff));
+        EXPECT_EQ(is_umr, nullptr != memh->exported_umr_mkey);
+        EXPECT_EQ(is_umr, nullptr == memh->cross_mr);
+    }
+};
+
+UCS_TEST_P(test_devx_umr_mkey, lazy_init)
+{
+    check_umr_init(false, false);
+}
+
+UCS_TEST_P(test_devx_umr_mkey, export_mkey_no_xgvmi)
+{
+    uct_ib_mlx5_devx_mem_t *memh = create_memh(1024);
+
+    /* Drop XGVMI capability */
+    md()->flags &= ~UCT_IB_MLX5_MD_FLAG_INDIRECT_XGVMI;
+
+    ASSERT_UCS_OK(uct_ib_mlx5_devx_reg_exported_key(md(), memh));
+
+    /* UMR objects remains uninitialized, exported key is not UMR one */
+    check_umr_init(false, false);
+    check_memh_export(memh, false);
+    destroy_memh(memh);
+}
+
+UCS_TEST_P(test_devx_umr_mkey, export_mkey_xgvmi)
+{
+    if (!(md()->flags & UCT_IB_MLX5_MD_FLAG_INDIRECT_XGVMI)) {
+        UCS_TEST_SKIP_R("XGVMI is not supported");
+    }
+
+    uct_ib_mlx5_devx_mem_t *memh = create_memh(1024);
+    ASSERT_UCS_OK(uct_ib_mlx5_devx_reg_exported_key(md(), memh));
+
+    /* Check that XGVMI capability had not been dropped */
+    ASSERT_TRUE(md()->flags & UCT_IB_MLX5_MD_FLAG_INDIRECT_XGVMI);
+
+    /* UMR export transport objects must be initialized */
+    check_umr_init(true, false);
+    /* UMR mkey pool must be empty */
+    ASSERT_TRUE(ucs_list_is_empty(&md()->umr.mkey_pool));
+    check_memh_export(memh, true);
+    destroy_memh(memh);
+
+    /* After release of memh, associated UMR mkey is moved back to mkey pool */
+    ASSERT_EQ(1, ucs_list_length(&md()->umr.mkey_pool));
+}
+
+UCS_TEST_P(test_devx_umr_mkey, export_mkey_xgvmi_reuse)
+{
+    if (!(md()->flags & UCT_IB_MLX5_MD_FLAG_INDIRECT_XGVMI)) {
+        UCS_TEST_SKIP_R("XGVMI is not supported");
+    }
+
+    const size_t MEMH_COUNT = 10;
+    uct_ib_mlx5_devx_mem_t *memh[MEMH_COUNT];
+
+    for (size_t i = 0; i < MEMH_COUNT; ++i) {
+        memh[i] = create_memh(1024);
+        ASSERT_UCS_OK(uct_ib_mlx5_devx_reg_exported_key(md(), memh[i]));
+        check_memh_export(memh[i], true);
+    }
+
+    /* UMR mkey pool must be empty */
+    ASSERT_TRUE(ucs_list_is_empty(&md()->umr.mkey_pool));
+
+    for (size_t i = 0; i < MEMH_COUNT; ++i) {
+        destroy_memh(memh[i]);
+    }
+
+    /* After release all allocated UMR keys go back to the mkey pool */
+    ASSERT_EQ(MEMH_COUNT, ucs_list_length(&md()->umr.mkey_pool));
+
+    /* Now reuse the keys for different handles of different size */
+    const size_t REUSE_COUNT = 5;
+    for (size_t i = 0; i < REUSE_COUNT; ++i) {
+        memh[i] = create_memh(2048);
+        ASSERT_UCS_OK(uct_ib_mlx5_devx_reg_exported_key(md(), memh[i]));
+        check_memh_export(memh[i], true);
+    }
+
+    ASSERT_EQ(MEMH_COUNT - REUSE_COUNT, ucs_list_length(&md()->umr.mkey_pool));
+
+    for (size_t i = 0; i < REUSE_COUNT; ++i) {
+        destroy_memh(memh[i]);
+    }
+
+    ASSERT_EQ(MEMH_COUNT, ucs_list_length(&md()->umr.mkey_pool));
+}
+
+UCS_TEST_P(test_devx_umr_mkey, import_mkey_no_umr)
```

Comment:
seems that test is failing on `ALLOW_OTHER_VHCA_ACCESS` on non-supported xgvmi FW, called from `uct_ib_mlx5_devx_xgvmi_umem_mr()`.

We might need to fix it in separate PR with either enabling `silent` or removing `uct_ib_mlx5_devx_allow_xgvmi_access()` when non supported.